src/cascadia/TerminalSettingsModel/HashUtils.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/*++
+Module Name:
+- HashUtils.h
+
+Abstract:
+- This module is used for hashing data consistently
+
+Author(s):
+- Carlos Zamora (CaZamor) 15-Apr-2021
+
+Revision History:
+- N/A
+--*/
+
+#pragma once
+
+namespace Microsoft::Terminal::Settings::Model::HashUtils
+{
+    // This is a helper template function for hashing multiple variables in conjunction to each other.
+    template<typename T>
+    constexpr size_t HashProperty(const T& val)
+    {
+        std::hash<T> hashFunc;
+        return hashFunc(val);
+    }
+
+    template<typename T, typename... Args>
+    constexpr size_t HashProperty(const T& val, Args&&... more)
+    {
+        return HashProperty(val) ^ HashProperty(std::forward<Args>(more)...);

```

Comment:
you are going to want a better hash combining algorithm than this. consider stealing the one from boost's `hash_combine`. This is a dangerous game :)

Reply:
(we already ship boost, so there is no license change)

Reply:
The MSVC STL ships with FNV1a which is pretty alright for such a simplistic XOR combination. So at least for hash quality boost's `hash_combine` wouldn't be an improvement per se.
IT would break the symmetry of this hash function though. At the moment `HashProperty("foo", "bar")` and `HashProperty("bar", "foo")` return the same value which indeed might be an issue.